wordpress-mobile/WordPress-Android
# Issue/10646 material login flow

Fixes #10646 

This PR moves login flow to the material theme and adds Dark Mode support. 

When deciding on how to approach the styling of the library from the client, I decided to keep the styles overload approach we are using right now. It provides the most flexibility since you are not forced to implement anything. I added some additional styles, so we can control more elements. We are not overriding colors anymore.

Like with other Dark Mode PR's I'm improvising a lot, so feel free to point anything weird. ( eg. I'm not sure what to do with our "greyish" button that we use in a couple of places).

There is a good amount of screens to check, so I split them into couple of sections. I skipped screens that are really similiar:

**Sign-Up**

![comparison_login_signup](https://user-images.githubusercontent.com/728822/73722333-1c039380-46db-11ea-8158-8806344b593d.png)

**Login**

![comparison_login](https://user-images.githubusercontent.com/728822/73722370-2c1b7300-46db-11ea-8ea0-5279eb5e5792.png)

**Landscape layouts**

![comparison_login_landscape](https://user-images.githubusercontent.com/728822/73722402-405f7000-46db-11ea-8fa6-9648f4b8dd86.png)

Notes: 
[This](https://github.com/wordpress-mobile/WordPress-Android/pull/10279#issuecomment-572495028) known issue with new appcompat library is not addressed here.

Username picker/changed is updated in #11112.

To test:
- Go through various login and sign-up flows and make sure everything looks good in both light and dark modes.

PR submission checklist:
